src/Http/Headers/src/BaseHeaderParser.cs

```diff
@@ -12,11 +13,11 @@ protected BaseHeaderParser(bool supportsMultipleValues)
         {
         }
 
-        protected abstract int GetParsedValueLength(StringSegment value, int startIndex, out T parsedValue);
+        protected abstract int GetParsedValueLength(StringSegment value, int startIndex, [MaybeNull] out T parsedValue);
 
-        public sealed override bool TryParseValue(StringSegment value, ref int index, out T parsedValue)
+        public sealed override bool TryParseValue(StringSegment value, ref int index, [MaybeNull] out T parsedValue)
```

Comment:
Can `[NotNullWhen(true)]` be applied here too?

Reply:
Unfortunately not. The `SupportsMultipleValues` pattern makes nullable annotations really awkward to use .